### DISCUSSION TITLE
Is there a limit for embedded lists on Isar?

### DISCUSSION THREAD

**Discussion by Alexcbecker** [isar/isar#1313] [Help] [answered]
Hello!

I'm trying to generate the files for the following classes:

```
part 'asset_report_model.g.dart';

@Collection()
class AssetReportModel {
  Id id = Isar.autoIncrement;
  final String apiId;
  final Realty realty;
  final Customer customer;
  final Escort escort;
  final String reportType;
  final List<AssetReportAreaModel> areas = List.empty(growable: true);
}
```
```
part 'asset_report_area.g.dart';


@Embedded()
class AssetReportAreaModel {

  final String? id;
  final int? order;
  final String? title;
  final List<AssetReportFieldModel> fields = List.empty(growable: true);

}
```
```
part 'asset_report_field.g.dart';

@Embedded()
class AssetReportFieldModel {
  final String? id;
  final int? order;
  final String? name;
  @Enumerated(EnumType.name)
  final AssetReportFieldType? type;
  final AssetReportCatalog? catalog;
  final String? hint;
  final String? label;
}
```

```
@Embedded()
class AssetReportCatalog {

  final String? title;
  final String? desc;
  final List<AssetReportCatalogItem> items= List.empty(growable: true);
  final List<AssetReportFieldModel> fields= List.empty(growable: true);
}
```

The problem that I'm having is the following:

![image](https://github.com/isar/isar/assets/22579778/41ec9a55-464e-4844-af81-104d1a5e213f)

If I comment the line **final List<AssetReportFieldModel> fields = List.empty(growable: true);** in **AssetReportAreaModel** it works without a problem. 

Am I missing something here? Is there a limit for how many embedded lists I can have?

Thanks in advance!

**Comment by giga10** [comment]
Check this issue: https://github.com/isar/isar/issues/703

Maybe it is related.

**Reply by Alexcbecker** [reply]
It is not the same problem, but that is interesting. Thanks!

**Comment by zawadZwd** [comment]
I had similar issue some time ago, lint do not say about it, but probably you are missing import of one or more class.

**Reply by Alexcbecker** [reply]
But the problem is with the generated code, so even if it is the case, I can't edit it.

**Reply by zawadZwd** [reply]
Do you have import every embedded class in AssertReportModel file?

**Reply by Alexcbecker** [reply]
No, because the structure is AssetReportModel > AssetReportAreaModel > AssetReportFieldModel > AssetReportCatalog. The imports are inside the embedded lists that contains the other classes.

**Reply by zawadZwd** [reply, marked as answer]
Class with the collection annotation should has imports of all embedded classes. Eg.

![image](https://github.com/isar/isar/assets/24791996/e6f2561a-189e-4468-8477-f2a223fd1c55)

**Reply by Alexcbecker** [reply]
Thank you for your help!
